### PR TITLE
troubleshoot collector addition to Vendor CLI

### PR DESCRIPTION
**Pull request by smithers1221**

### CODE REVIEW

**Comment by smithers1221** [comment]
[ch20135]

File: pkg/platformclient/collector.go [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+
+type PlatformChannel struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+func (c *HTTPClient) ListCollectors(appID string, appType string) ([]v1.AppCollectorInfo, error) {
+	response := GraphQLResponseListCollectors{}
+
+	request := graphql.Request{
```
Comment (divolgin) [inline pkg/platformclient/collector.go]: There shouldn't be any GQL requests in this package.

File: client/collector.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	"github.com/replicatedhq/replicated/pkg/types"
+)
+
+func (c *Client) ListCollectors(appID string, appType string) ([]types.CollectorInfo, error) {
+	appType, err := c.GetAppType(appID)
+	if err != nil {
+		return nil, err
+	}
+
+	if appType == "platform" {
```
Comment (divolgin) [inline client/collector.go]: both `ListCollectors` functions are identical. You don't need this split logic here.

File: client/collector.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+		}
+
+		return shipCollectorInfos, nil
+	}
+
+	return nil, errors.New("unknown app type")
+}
+
+func (c *Client) UpdateCollector(appID string, appType string, specID string, yaml string) (interface{}, error) {
+	if appType == "platform" {
```
Comment (divolgin) [inline client/collector.go]: both `UpdateCollector` functions are identical.

File: client/collector.go [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+		return c.ShipClient.GetCollector(appID, appType, specID)
+	}
+
+	return nil, errors.New("unknown app type")
+
+}
+
+func (c *Client) PromoteCollector(appID string, appType string, specID string, channelIDs ...string) error {
+
+	if appType == "platform" {
```
Comment (divolgin) [inline client/collector.go]: This is the only functions that needs to make distinction between platform and ship apps

**Pull request closed without merging**